phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolForDeleteBeforeRebuildIT.java

```diff
@@ -0,0 +1,217 @@
+        viewFullName = SchemaUtil.getTableName(schemaName, viewName);
+        globalIndexName = "GLBL_IDX_" + generateUniqueName();
+        globalIndexFullName = SchemaUtil.getTableName(schemaName, globalIndexName);
+        createTestTable(getUrl(), String.format(DATA_TABLE_DDL, dataTableFullName));
+        createTestTable(getUrl(), String.format(VIEW_DDL, viewFullName, dataTableFullName));
+        String dataTableUpsert = String.format(UPSERT_SQL, dataTableFullName);
+        PreparedStatement stmt = conn.prepareStatement(dataTableUpsert);
+        for (int i=1; i < 4; i++) {
+            upsertRow(stmt, "tenantID1", i, "name" + i, 9990+i);
+        }
+        conn.commit();
+    }
+
+    private void upsertRow(PreparedStatement stmt, String tenantId, int id, String name, int zip)
+            throws SQLException {
+        int index = 1;
+        stmt.setString(index++, tenantId);
+        stmt.setInt(index++, id);
+        stmt.setString(index++, name);
+        stmt.setInt(index++, zip);
+        stmt.executeUpdate();
+    }
+
+    @After
+    public void teardown() throws SQLException {
+        if (conn != null) {
+            conn.close();
+        }
+    }
+
+    @Test
+    /**
+     * IndexTool should return -1 for View Indexes because view indexes share the same table.
+     */
+    public void testDeleteBeforeRebuildForViewIndexShouldFail() throws Exception {
+        String createViewIndex = String.format(INDEX_GLOBAL_DDL, globalIndexName, viewFullName);
+        PreparedStatement stmt = conn.prepareStatement(createViewIndex);
+        stmt.execute();
+        runIndexTool(schemaName, viewName, globalIndexName, -1);
+    }
+
+    @Test
+    /**
+     * Test delete before rebuild
+     */
+    public void testDeleteBeforeRebuildForGlobalIndex() throws Exception {
+        conn.createStatement().execute(String.format(INDEX_GLOBAL_DDL, globalIndexName, dataTableFullName));
+        String globalIndexUpsert = String.format(UPSERT_SQL, globalIndexFullName);
+        PreparedStatement stmt = conn.prepareStatement(globalIndexUpsert);
+        upsertRow(stmt, "tenantID1",11, "name11", 99911);
+        conn.commit();
+
+        ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
+        PTable physicalTable = PhoenixRuntime.getTable(conn, globalIndexFullName);
+        Table hIndexTable= queryServices.getTable(physicalTable.getPhysicalName().getBytes());
+        int count = getUtility().countRows(hIndexTable);
+        // Confirm index has rows.
+        assertEquals(4, count);
+
+        runIndexTool(schemaName, dataTableName, globalIndexName, 0);
+
+        count = getUtility().countRows(hIndexTable);
+
+        // Confirm index has all the data rows
+        assertEquals(3, count);
+    }
+
+    @Test
+    /**
+     * For local indexes the data is on the same row and all local indexes share the same column family
+     * So, it should return -1 for local indexes.
```

Comment:
Nit: If you are close to being done for this PR, we can defer this for later. Perhaps we can replace 0 and -1's with consts/enums with names to indicate SUCCESS/FAILURE (or potentially specific failures in the future.

Reply:
The index tool is designed to also be runnable from the shell so the integer-based codes are meant to play nicely with Unix.